RealmSwift/AnyRealmValue.swift

```diff
@@ -156,4 +178,59 @@ public enum AnyRealmValue: Hashable {
     public init() {
         self = .none
     }
+
+    /// Returns a `AnyRealmValue` storing a `Map`.
+    ///
+    /// - Parameter dictionary: A Swift's dictionary of `AnyRealmValue` values.
+    /// - Returns: Returns an `AnyRealmValue` storing a `Map`.
+    public static func fromDictionary(_ dictionary: Dictionary<String, AnyRealmValue>) -> AnyRealmValue {
+        let map = Map<String, AnyRealmValue>()
+        map.merge(dictionary, uniquingKeysWith: { $1 })
+        return AnyRealmValue.dictionary(map)
+    }
+
+    /// Returns a `AnyRealmValue` storing a `List`.
+    ///
+    /// - Parameter array: A Swift's array of `AnyRealmValue`.
+    /// - Returns: Returns a `AnyRealmValue` storing a `List`.
+    public static func fromArray(_ array: Array<AnyRealmValue>) -> AnyRealmValue {
+        let list = List<AnyRealmValue>()
+        list.append(objectsIn: array)
+        return AnyRealmValue.list(list)
+    }
+
+    // MARK: - Hashable
+
+    public func hash(into hasher: inout Hasher) {
+        switch self {
+        case let .int(i):
+            hasher.combine(i.hashValue)
```

Comment:
These should all be `hasher.combine(i)` without the `.hashValue`. This currently is hashing the values and then rehashing the hash.